R/deployApp.R

```diff
@@ -336,12 +336,13 @@ deployApp <- function(appDir = getwd(),
   accountDetails <- accountInfo(target$account, target$server)
 
   # test for compatibility between account type and publish intent
-  if (isShinyapps(accountDetails$server)) {
-    # ensure we aren't trying to publish an API to shinyapps.io; this will not
-    # currently end well
-    if (identical(contentCategory, "api")) {
-      stop("Plumber APIs are not currently supported on shinyapps.io; they ",
-           "can only be published to RStudio Connect.")
+  if (isCloudServer(accountDetails$server)) {
```

Comment:
This can probably just check shinyapps directly?

```
if (isShinyappsServer(accountDetails$server)) {
  if (identical(contentCategory, "api")) {
    stop("Plumber APIs are not currently supported on shinyapps.io; they ",
         "can only be published to RStudio Connect or rstudio.cloud.")
  }
}
```

Reply:
We should never be relying a `contentCategory` of `api`. I thought we had ditched that.

It was removed with https://github.com/rstudio/rsconnect/pull/304

Unfortunately, it returned with the fix to https://github.com/rstudio/rsconnect/issues/204 (https://github.com/rstudio/rsconnect/commit/9e768e87c9a63885f175ccbc4ed69eb6c503f0f2; no PR?)

~This code is after we call `inferAppMode` and can use `appMode` rather than `contentCategory`.~

We do not want `deployAPI` to set `contentCategory`.

Correction: This check is happening ahead of the call to `bundleApp`, which means that we have not yet determined the `appMode`. That's unfortunate.